### PR TITLE
chore(datadog sinks)!: extract `default_api_key` into shared config struct

### PR DESCRIPTION
**Pull request by neuronull**
Addresses the TODO indroduced by https://github.com/vectordotdev/vector/pull/16280

, now that the serde alias is compatible with flatten.

### CODE REVIEW

**Comment by spencergilbert** [comment]
Is there a test for this behavior, just to ensure the previous behavior is still adhered to?

**Comment by neuronull** [comment]
>  Is there a test for this behavior, just to ensure the previous behavior is still adhered to?

The performance test that failed CI in the linked PR

**Comment by spencergilbert** [comment]
A unit test might be nice? `*Config::from_toml` sort of thing?

**Comment by neuronull** [comment]
Ah. The unit tests that are present in the existing DD sinks cover the functionality of the `default_api_key` being moved into the shared structure.

idk if there is much benefit in adding a separate unit test to validate the `api_key` alias, since we want to remove that in v0.29.0 anyway.

as part of my intent to make this a Draft PR, I was going to flag a comment to that effect, asking if its ok to remove that alias as part of this PR.

File: src/sinks/datadog/events/config.rs [hunk 1]
```diff
@@ -105,11 +92,9 @@ impl DatadogEventsConfig {
...
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
         let client = self.build_client(cx.proxy())?;
-        let healthcheck = self.dd_common.build_healthcheck(
-            client.clone(),
-            self.default_api_key.clone().into(),
-            self.region.as_ref(),
-        )?;
+        let healthcheck = self
+            .dd_common
+            .build_healthcheck(client.clone(), self.region.as_ref())?;
```
Comment (bruceg) [inline src/sinks/datadog/events/config.rs:97]: Would it make sense to take this a step further and move `region` into `dd_common` as well?

File: website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+authors: ["neuronull"]
+release: "0.29.0"
+hide_on_release_notes: false
+badges:
+  type: breaking change
+---
+
+Vector's 0.29.0 release includes **deprecations**:
+
+1. [The `datadog` sink has been renamed to `mezmo`](#mezmo_sink)
```
Comment (bruceg) [inline website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md:14]: Um, the `datadog` sink? 😕

File: website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (bruceg) [inline website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md:1]: Adding this file seems unrelated.

**Review by bruceg — COMMENTED** [review]
Code looks reasonable, but what is the upgrade highlight for `mezmo` doing here?

File: website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md (hunk 2 above)
Comment (neuronull) [reply website/content/en/highlights/2023-04-11-0-29-0-upgrade-guide.md]: accidentally had the file in read only mode so my edits didnt get in 🤦

**Comment by neuronull** [comment]
> Code looks reasonable, but what is the upgrade highlight for `mezmo` doing here?

My save of the edits didn"t take (stole the upgrade guide from Stephen's PR which isn't merged yet, and editted it).

The reason for the breaking change is due to the change in https://github.com/vectordotdev/vector/pull/16731/commits/c5623e571e6db521cad99d6b71496bae34f78f17

File: src/sinks/datadog/events/config.rs (hunk 1 above)
Comment (neuronull) [reply src/sinks/datadog/events/config.rs:97]: Ah, I had evaluated that on the initial PR for the shared struct, but the `region` is not part of the `datadog_traces` config. (region is considered deprecated and replaced by `site`)

**Comment by neuronull** [comment]
*sob* all this cause my selection of draft PR from the drop down somehow didn't take. I blame jumpy mouse. Sorry for the thrash, I really did try to open this as a draft. AFAICT you can't move it out to draft after its been "ready".

**Comment by jszwedko** [comment]
> _sob_ all this cause my selection of draft PR from the drop down somehow didn't take. I blame jumpy mouse. Sorry for the thrash, I really did try to open this as a draft. AFAICT you can't move it out to draft after its been "ready".

You can actually convert it back to a draft. The link is under the reviewers on the right (search for "Convert to draft").

**Comment by neuronull** [comment]
> You can actually convert it back to a draft. The link is under the reviewers on the right (search for "Convert to draft").

Ohhh I see that now thank you! I will remember that going forward.

**Comment by neuronull** [comment]
I'll close this and open a new one against master since Bruce's PR was closed
